### DISCUSSION TITLE
Intel(R) Celeron(R) N5105 cannot boot from iSCSI

### DISCUSSION THREAD

**Discussion by wxl2** [ipxe/ipxe#616] [Q&A]
I encountered some issues while using ipxe to boot an iscsi image using Intel Celeron N5105.  When I used Intel Celeron J4125, it could boot successfully.  The error is as follows:

input output i/o error
san close time
![微信图片_20220316122530](https://user-images.githubusercontent.com/45157566/158516699-e02bc9d3-2b60-4ea7-a009-3221c85f450e.jpg)
out
![微信图片_20220316122521](https://user-images.githubusercontent.com/45157566/158516712-87972b90-7d5c-4307-bd39-d1291dbe2ad3.jpg)

**Comment by NiKiZe** [comment]
Can you test that data fetching works correctly? For example http://boot.ipxe.org/demo

Verify server logs, and maybe also check tcpdump or similar.

**Reply by wxl2** [reply]
When the linux OS was installed, it was found that it could get part of the initrd file, and then it would be stuck and cannot get any more contents from the iscsi target. The ipxe script is as follows:

set root-path iscsi:192.168.99.150:::1:wxl.2022-03.os.ipxe:storage.cs
set keep-san 1
sanhook ${root-path}
set server_root http://172.20.1.67/centos/
kernel ${server_root}/images/pxeboot/vmlinuz initrd=initrd.img inst.repo=${server_root}
initrd ${server_root}/images/pxeboot/initrd.img
boot || goto start

**Reply by stappersg** [reply]
Regarding the _Verify server logs, and maybe also check tcpdump or similar_: Do it first for the working J4125 so you have a better understanding where the N5105 fails.

**Reply by NiKiZe** [reply]
So is that issue without iscsi as well?

And comparing output to the working setup is only needed if there is nothing obvious. Don't waste your time on that.

@stappersg I suggest that you stop taking my words and make it into something else. JUST STOP!

**Reply by wxl2** [reply]
I downloaded the ISO from the official website of IPXE and burned it into a USB disk. The system can be normally booted by using the USB disk, but using the compiled snponly.efi file the system cannot be normally booted. My compilation parameter for snponly.efi file is:  make bin-x86_64-efi/snponly.efi EMBED=boot.ipxe -j4  DEBUG=iscsi,scsi,efi_block,sanboot     

It is suspicious that the parameter is wrong somewhere which made the difference from the official ISO. Could you show me some guidance over that? Thanks for the advice!

**Reply by stappersg** [reply]
@wxl advice keep verifing that each change you make in the configuration keeps working on the `J4125`, for better isolating the problem to the `N5105`

**Reply by wxl2** [reply]
I solved this problem by changing the compilation method

**Reply by stappersg** [reply]
Karma points for sharing the compilation method

**Comment by stappersg** [comment]
Screenshot:
![afbeelding](https://user-images.githubusercontent.com/35434094/159011989-5dbb07d0-c408-43ee-a9b0-28675b759325.png)

At the time of writing this update there was a two day silence.
It still would be nice if the known working compilation method is shared in this _github.com discussion_.

It is not important that I think that @wxl2 had a working iPXE from the begin of this thread.

Important, in the _iPXE universum_, is that iPXE works and we find ways to even improve it.


At `NiXZe` acknowledge on your _I feel offended, I feel the need to defend myself_.
Your _stop request_ is also recieved, it will not be granted.
Thing I can do slash should do is transmitting more clearly _reporter has a working iPXE_ when reporter has a working iPXE.


Regards
Stappers

P.S.
I'm fine with both this discussion marked as **answered** and **left unmarked** of being answered.
